server/http.go

```diff
@@ -915,3 +984,45 @@ func (p *Plugin) slackAttachmentToUpdatePMI(currentValue, channelID string) *mod
 
 	return &slackAttachment
 }
+
+func (p *Plugin) checkChannelPreference(channelID string) (bool, int, error) {
+	channel, appErr := p.API.GetChannel(channelID)
+	if appErr != nil {
+		return false, http.StatusInternalServerError, errors.New(appErr.Message)
+	}
+
+	zoomChannelSettingsMap, err := p.listZoomChannelSettings()
+	if err != nil {
+		return false, http.StatusInternalServerError, err
+	}
+
+	val, exist := zoomChannelSettingsMap[channelID]
+	preference := false
+	/*
+		Check if zoom settings for current channel exist.
+		Check if creating meeting is disabled in the plugin configuration.
+	*/
+	if exist {
+		if val.Preference == ZoomChannelPreferences[DefaultPreference] {
+			preference = p.configuration.RestrictMeetingCreation
+		} else if val.Preference == ZoomChannelPreferences[EnablePreference] {
+			preference = true
+		}
+	} else if channel.Type == model.ChannelTypeOpen {
+		preference = p.configuration.RestrictMeetingCreation
+	}
+
+	return preference, http.StatusOK, nil
+}
+
+func (mv ZoomChannelSettingsMapValue) IsValid() error {
+	if mv.ChannelName == "" {
+		return errors.New("channel name should not be empty")
+	}
+
+	if mv.Preference == "" {
+		return errors.New("preference should not be empty")
+	}
```

Comment:
This needs to match a specific set of pre-defined values, so we should check that here